src/platform/site-wide/component-library-analytics-setup.js

```diff
@@ -275,20 +275,55 @@ const analyticsEvents = {
       action: 'change',
       event: 'int-checkbox-option-click',
       prefix: 'checkbox',
+      ga4: {
+        event: 'interaction',
+        component_name: 'va-checkbox',
+        custom_string_1: 'component-library',
+        mapping: {
+          'checkbox-label': 'heading_1',
+          'checkbox-description': 'heading_2',
+          'checkbox-required': 'required',
+          'checkbox-checked': 'value',
+          version: 'component_version',
+        },
+      },
     },
   ],
   'va-checkbox-group': [
     {
       action: 'change',
       event: 'int-checkbox-group-option-click',
       prefix: 'checkbox-group',
+      ga4: {
+        event: 'interaction',
+        component_name: 'va-checkbox-group',
+        custom_string_1: 'component-library',
+        mapping: {
+          'checkbox-group-label': 'heading_1',
+          'checkbox-group-optionLabel': 'custom_string_2',
+          'checkbox-group-required': 'required',
+          version: 'component_version',
+        },
+      },
     },
   ],
   'va-date': [
     {
       action: 'blur',
       event: 'int-date-blur',
       prefix: 'date',
+      ga4: {
+        event: 'interaction',
+        component_name: 'va-date',
+        custom_string_1: 'component-library',
+        mapping: {
+          'date-year': 'value',
+          'date-month': 'custom_number_1',
+          'date-day': 'custom_number_2',
+          'date-month-year-only': 'status',
```

Comment:
I'm not sure about the mappings for these. If month and day are custom numbers, why is `date-year` value? I'm also not sure what "status" means in the context of "date-month-year-only".

```
'date-year': 'value',
'date-month-year-only': 'status',
```

We may need to connect with Michelle about the best way to map these values.

Reply:
For the year, month, and day fields, I chose to use `value`, `custom_number_1`, and `custom_number_2` because we emit three user-entered values, thus I need to find three GA actions that can correspond to these while making sense. `value` seems like the best choice since it's user inputted data, but there aren't more value fields (i.e. `value_1`, `value_2`, etc.), so I looked at other GA events that would make sense as a user value. The `custom_number` fields seemed to be the best fit. I do agree that it would be more consistent to use three `value` or `custom_number` fields, but I didn't think that's an option since it isn't listed on the GA4 planning doc.

As for the `date-month-year-only`, this is a boolean that modifies which fields we show to the user. I thought that either `status` or `type` would make the most sense, but I went with `status` since that sounded like the best way to indicate that we're indicating a specific state of the date component.

I am absolutely open to changing any of these if there are better options. Let me know what you think, and in the meantime, I'm going to reach out to the Analytics and Insight support team soon and see what they think too.